worldengine/hdf5_serialization.py

```diff
@@ -0,0 +1,220 @@
+
+    if hasattr(world, 'temperature'):
+        temperature_grp = f.create_group("temperature")
+        temperature_ths_grp = temperature_grp.create_group("thresholds")
+        temperature_ths_grp['polar'] = world.temperature['thresholds'][0][1]
+        temperature_ths_grp['alpine'] = world.temperature['thresholds'][1][1]
+        temperature_ths_grp['boreal'] = world.temperature['thresholds'][2][1]
+        temperature_ths_grp['cool'] = world.temperature['thresholds'][3][1]
+        temperature_ths_grp['warm'] = world.temperature['thresholds'][4][1]
+        temperature_ths_grp['subtropical'] = world.temperature['thresholds'][5][1]
+        temperature_data = temperature_grp.create_dataset("data", (world.height, world.width), dtype=numpy.float)
+        temperature_data.write_direct(world.temperature['data'])
+
+    # lake_map and river_map have inverted coordinates
+    if hasattr(world, 'lake_map'):
+        lake_map_data = f.create_dataset("lake_map", (world.height, world.width), dtype=numpy.float)
+        lake_map_data.write_direct(world.lake_map)
+
+    # lake_map and river_map have inverted coordinates
+    if hasattr(world, 'river_map'):
+        river_map_data = f.create_dataset("river_map", (world.height, world.width), dtype=numpy.float)
+        river_map_data.write_direct(world.river_map)
+
+    generation_params_grp = f.create_group("generation_params")
+    generation_params_grp['seed'] = world.seed
+    generation_params_grp['n_plates'] = world.n_plates
+    generation_params_grp['ocean_level'] = world.ocean_level
+    generation_params_grp['step'] = world.step.name
+
+    f.close()
+
+
+def _from_hdf5_quantiles(p_quantiles):
+    quantiles = {}
+    for p_quantile in p_quantiles:
+        quantiles[p_quantile.title()] = p_quantiles[p_quantile].value
+    return quantiles
+
+
+def _from_hdf5_matrix_with_quantiles(p_matrix):
+    matrix = dict()
+    matrix['data'] = p_matrix['data']
+    matrix['quantiles'] = _from_hdf5_quantiles(p_matrix['quantiles'])
+    return matrix
+
+
+def load_world_to_hdf5(filename):
+    f = h5py.File(filename, libver='latest', mode='r')
+
+    w = World(f['general/name'].value,
+              f['general/width'].value,
+              f['general/height'].value,
+              f['generation_params/seed'].value,
+              f['generation_params/n_plates'].value,
+              f['generation_params/ocean_level'].value,
+              Step.get_by_name(f['generation_params/step'].value))
+
+    # Elevation
+    e = numpy.array(f['elevation/data'])
+    e_th = [('sea', f['elevation/thresholds/sea'].value),
+            ('plain', f['elevation/thresholds/plain'].value),
+            ('hill', f['elevation/thresholds/hill'].value),
+            ('mountain', None)]
+    w.set_elevation(e, e_th)
+
+    # Plates
+    w.set_plates(numpy.array(f['plates']))
+
+    # Ocean
+    w.set_ocean(numpy.array(f['ocean']))
+    w.sea_depth = numpy.array(f['sea_depth'])
+
+    # Biome
+    if 'biome' in f.keys():
```

Comment:
well, it should not cause a crash: the functions using optional data should first check if that data is available. For example all the simulations do that:

```
class ErosionSimulation(object):
    def is_applicable(self, world):
        return world.has_precipitations()
```